osconfig_tests/test_config/project_config.go

```diff
@@ -14,23 +14,64 @@
 
 package testconfig
 
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"sync"
+)
+
 // Project is details of test Project.
 type Project struct {
 	TestProjectID        string
 	ServiceAccountEmail  string
-	TestZone             string
 	ServiceAccountScopes []string
+	testZones            map[string]int
+	zoneIndices          map[int]string
+	mux                  sync.Mutex
 }
 
 // GetProject ...
-func GetProject(projectID, testZone string) *Project {
+func GetProject(projectID string, testZones *map[string]int) *Project {
+	zoneIndices := make(map[int]string, len(*testZones))
+
+	i := 0
+	for z := range *testZones {
+		zoneIndices[i] = z
+		i++
+	}
+
 	return &Project{
 		TestProjectID:       projectID,
-		TestZone:            testZone,
+		testZones:           *testZones,
+		zoneIndices:         zoneIndices,
 		ServiceAccountEmail: "default",
 		ServiceAccountScopes: []string{
 			"https://www.googleapis.com/auth/cloud-platform",
 			"https://www.googleapis.com/auth/devstorage.full_control",
 		},
 	}
 }
+
+// GetZone gets a random zone that still has capacity.
+func (p *Project) GetZone() string {
+	p.mux.Lock()
+	defer p.mux.Unlock()
+
+	zc := len(p.zoneIndices)
+	if zc == 0 {
+		fmt.Println("Not enough zone quota sepcified. Specify additional quota in `test_zones`.")
```

Comment:
This should be returned as an error. To make this make more sense CreateComputeInstance should probably just take testProjectConfig and handle all the logic in that function. Also the way this is currently setup makes it very hard to add logic to just wait for an open slot. 

Since this would all take a large refactor a TODO would probably be ok for now.